src/common/transformations/src/transformations/common_optimizations/shuffle_channels_fusion.cpp

```diff
@@ -98,22 +98,21 @@ ov::pass::ShuffleChannelsFusion::ShuffleChannelsFusion(const bool reshape_consta
         const auto& pattern_map = m.get_pattern_value_map();
 
         auto data = pattern_map.at(input);
+        auto reshape_before_constant = std::dynamic_pointer_cast<opset6::Constant>(
+            pattern_map.at(reshape_before_const_pattern).get_node_shared_ptr());
         auto reshape_before =
             std::dynamic_pointer_cast<opset6::Reshape>(pattern_map.at(reshape_before_pattern).get_node_shared_ptr());
         auto transpose =
             std::dynamic_pointer_cast<opset6::Transpose>(pattern_map.at(transpose_pattern).get_node_shared_ptr());
         auto reshape_after =
             std::dynamic_pointer_cast<opset6::Reshape>(pattern_map.at(reshape_after_pattern).get_node_shared_ptr());
+        auto reshape_after_constant = std::dynamic_pointer_cast<opset6::Constant>(
```

Comment:
if `dynamic_pointer_case` returns `nullptr` here, `copy_runtime_info` will throw an exception?

Reply:
> if `dynamic_pointer_case` returns `nullptr` here, will `copy_runtime_info` throw an exception?

I moved check for reshape_before_constant and reshape_after_constant.